crates/stackable-operator/src/time/duration.rs

```diff
@@ -301,6 +301,14 @@ impl Div<u32> for Duration {
     }
 }
 
+/// This implementation is needed for `CommonConfiguration::min_secret_lifetime`.
+/// The default value is arbitrary and should probably not be assumed at call sites.
```

Comment:
The default impl is needed to satisfy the `#[derive(Default)]` of `CommonConfiguration` which is independent of `serde` actions. I don't think there is an alternative.

Reply:
Okay that makes sense.

On the other hand, do we even need the `Default` impl. I can only see it used in unit tests (in this crate) and it is not used in the Airflow operator for example. But that is a fairly limited scope, so there might be places where it is used.

To keep the `Default` impl, we could use a manual implementation instead of the derived one. This would allow us to get rid of the arbitrary `Default` impl for `Duration`. I could look like this:

```rust
impl<T> Default for CommonConfiguration<T> where T: Default {
    fn default() -> Self {
        Self {
            min_secret_lifetime: Duration::from_secs(0),
            ..Default::default()
        }
    }
}
```

_(Untested, some small details could be missing)_

Reply:
I like this. If only we had a monorepo to quickly test it across all operators ;)

Reply:
[This](https://github.com/stackabletech/operator-rs/pull/908/commits/d67e5a08064d0b5da39346da298c34e90d5071fa) variant seems to work. Tested it with several ops. Thanks!

Reply:
Nice, happy to help!